pkg/pillar/cmd/zedmanager/updatestatus.go

```diff
@@ -176,15 +176,25 @@ func doInstall(ctx *zedmanagerContext,
 		log.Warnln(errString)
 	}
 
-	// If we are purging and we failed to activate due some volumes
-	// which are now removed from VolumeRefConfigList we remove them
-	if status.PurgeInprogress == types.RecreateVolumes && !status.Activated {
+	// Removing VolumeRefConfig which are removed from the AppInstanceConfig
+	// and not used by any domain while purging. VolumeRefConfig which are
+	// not in AppInstanceConfig but used by some running domain will be
+	// removed as part of purgeCmdDone.
+	if status.PurgeInprogress == types.RecreateVolumes {
+		domainVolMap := make(map[string]int)
```

Comment:
Nit but the idiom for what you're doing is a map[string]bool since you are just collecting they keys and not counting. Thus domainVolMap[dc.VolumeKey] = true

Reply:
Resolved